### PR TITLE
start_discovery: handle managed-dns-domains

### PR DESCRIPTION
**Pull request by danielerez**
In order to ensure registration of unique DNS records,
the cluster name should be randomly generated.
Hence, passed --managed-dns-domains to start_discovery.py
as an indication. Also, parsed the base domain from
args.managed_dns_domains map.

### CODE REVIEW

**Comment by danielerez** [comment]
@tsorya @romfreiman

File: discovery-infra/start_discovery.py [hunk 1]
```diff
@@ -311,6 +314,9 @@ def main():
         type=str,
         default="redhat.com",
     )
+    parser.add_argument(
+        "-mD", "--managed-dns-domains", help="Managed DNS domains", type=str, default=""
```
Comment (romfreiman) [inline discovery-infra/start_discovery.py:318]: cam u maybe write more descriptive help here. Not sure that I undestand what should be provided. Even give an example

Comment (danielerez) [reply discovery-infra/start_discovery.py]: sure, added explanation and example, also updated the readme in: https://github.com/openshift/assisted-test-infra/pull/16

Comment (danielerez) [reply discovery-infra/start_discovery.py]: added DNS validation in: https://github.com/openshift/assisted-test-infra/pull/21

**Pull request merged**